src/ncp/ncp_base.cpp

```diff
@@ -414,8 +416,8 @@ void NcpBase::HandleSendDone()
     if (mSendQueue.GetHead() != NULL)
     {
         Message &message(*mSendQueue.GetHead());
-        HandleDatagramFromStack(message);
         mSendQueue.Dequeue(message);
+        HandleDatagramFromStack(message);
```

Comment:
Yes, Dequeue does not free the message, it just removes it from the queue.

It looks like `HandleDatagramFromStack` is responsible for taking a message buffer and sending it to the host using `spinel`.  That is the behavior we want, right?